test/NLog.Extensions.Hosting.Tests/ExtensionMethodTests.cs

```diff
@@ -38,23 +38,48 @@ public void UseNLog_withConfiguration_WorksWithNLog()
             TestHostingResult(actual);
         }
 
-        private static void TestHostingResult(IHost host)
+        [Fact]
+        public void AddNLog_withShutdownOnDispose_worksWithNLog()
+        {
+            var someParam = new NLogProviderOptions { ShutdownOnDispose = true };
+            var actual = new HostBuilder().ConfigureLogging(l => l.AddNLog(someParam)).Build();
+            try
+            {
+                TestHostingResult(actual, false);
+                Assert.NotNull(LogManager.Configuration);
+            }
+            finally
+            {
+                actual.Dispose();
+                Assert.Null(LogManager.Configuration);
+            }
+        }
+
+        [Fact]
+        public void UseNLog_withAddNLog_worksWithNLog()
+        {
+            var actual = new HostBuilder().UseNLog().ConfigureServices((h,s) => s.AddLogging(l => l.AddNLog())).Build();
+            TestHostingResult(actual);
+        }
+
+        private static void TestHostingResult(IHost host, bool autoReset = true)
```

Comment:
small suggestion: to prevent refactor / merge / overload issues issues in future, I try to not use optional parameters for non-public methods nowadays. What do you think?

Reply:
Only have an opinion about public interfaces, where default-parameters are not good. Pretty loose when it comes to unit-testing.

Reply:
Have removed the default parameter, so everyone can have a good night sleep :)

Reply:
I learned, value test code as production code ;)

(but OK, with time constraints and tradeoffs between writing fixes/features and test code, that's hard or maybe even bad)